cuda/distributed/vector_kernels.cu

```diff
@@ -5,6 +5,9 @@
 #include "core/distributed/vector_kernels.hpp"
 
 
+#include <functional>
```

Comment:
why is this necessary? The kernel doesn't use `std::function`, and I would have guessed that `thrust/functional.h` would be more appropriate.

Reply:
Doesn't seem to be necessary, it was only added to make HIP and CUDA consistent